From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrei Bacescu <9010245+bacescuandrei@users.noreply.github.com>
Date: Tue, 27 Oct 2026 23:41:46 +0200
Subject: [PATCH] 10044: PowerPC: divw. should not update XER[OV] and XER[SO]

PowerPC: divw. does not update XER[OV] and XER[SO]

divw. has OE=0, so like divwu. it only sets CR0. It called divOverflow,
which set OV and SO on a zero divisor or 0x80000000 / -1 and carried the
bogus SO into CR0.
---
 Ghidra/Processors/PowerPC/data/languages/ppc_instructions.sinc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Ghidra/Processors/PowerPC/data/languages/ppc_instructions.sinc b/Ghidra/Processors/PowerPC/data/languages/ppc_instructions.sinc
index 3ce82a5cae..9574da8659 100644
--- a/Ghidra/Processors/PowerPC/data/languages/ppc_instructions.sinc
+++ b/Ghidra/Processors/PowerPC/data/languages/ppc_instructions.sinc
@@ -985,11 +985,9 @@
 :divw. D,A,B		is OP=31 & D & A & B & OE=0 & XOP_1_9=491 & Rc=1
 {
 @ifdef BIT_64
-	divOverflow(A:4,B:4);
 	D = sext(A:4 s/ B:4);
 	cr0flags(D:4);
 @else
-	divOverflow(A,B);
 	D = A s/ B;
 	cr0flags(D);
 @endif
-- 
2.45.1

